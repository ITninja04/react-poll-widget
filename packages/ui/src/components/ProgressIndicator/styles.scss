@import "../../colors";
@import "../../mixin";

.progress-indicator {
  width: 24px;
  height: 24px;

  svg {
    width: 100%;
    height: 100%;
    circle {
      fill: none;
      stroke-width: 2;
      stroke-linecap: round;

      &:first-child {
        stroke: rgba($white, 0.25);
      }

      &:last-child {
        transition: $transition-duration;
        stroke: $white;
        transform-origin: 12px 12px;
        transform: rotateZ(-90deg);
      }
    }
  }
}
