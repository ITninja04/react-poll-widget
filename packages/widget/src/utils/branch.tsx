export type BranchFunction = (condition: boolean, ComponentIfTrue: React.ReactNode, ComponentIfFalse: React.ReactNode) => React.ReactNode;

export const branch: BranchFunction = (condition, ComponentIfTrue, ComponentIfFalse) => condition
  ? ComponentIfTrue
  : ComponentIfFalse;