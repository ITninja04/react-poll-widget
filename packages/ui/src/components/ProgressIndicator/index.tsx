import React from 'react';

import './styles.scss';

const PI = 3.1415;
const R = 10;
const ProgressIndicator: React.FC<{
    step: number,
    stepsTotal: number
}> = ({
    step,
    stepsTotal
}) => (
            <div className="progress-indicator">
                <svg>
                    <circle
                        cx='12'
                        cy='12'
                        r={R}
                    />
                    <circle
                        cx='12'
                        cy='12'
                        r={R}
                        stroke-dasharray={2 * PI * R}
                        stroke-dashoffset={2 * PI * R * (1 - (step / stepsTotal))}
                    />
                </svg>
            </div>
        );

export default ProgressIndicator;