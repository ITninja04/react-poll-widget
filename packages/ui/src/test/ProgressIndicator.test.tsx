import ProgressIndicator from "../components/ProgressIndicator";
import { createSnapshotTest } from "./utils";

createSnapshotTest(ProgressIndicator, {}, 'Empty');
createSnapshotTest(ProgressIndicator, {
  step: 1,
  stepsTotal: 2
}, 'Half');