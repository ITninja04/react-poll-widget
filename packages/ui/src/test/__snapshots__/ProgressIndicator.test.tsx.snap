// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Empty should render 1`] = `
<DocumentFragment>
  <div
    class="progress-indicator"
  >
    <svg>
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <circle
        cx="12"
        cy="12"
        r="10"
        stroke-dasharray="62.830000000000005"
        stroke-dashoffset="NaN"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`(Snapshot) Half should render 1`] = `
<DocumentFragment>
  <div
    class="progress-indicator"
  >
    <svg>
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <circle
        cx="12"
        cy="12"
        r="10"
        stroke-dasharray="62.830000000000005"
        stroke-dashoffset="31.415000000000003"
      />
    </svg>
  </div>
</DocumentFragment>
`;
